src/doc/rustc/src/platform-support/arm-none-eabi.md

```diff
@@ -69,16 +140,23 @@ build-std = ["core"]
 ```
 
 Most of `core` should work as expected, with the following notes:
-* If the target is not `*hf`, then floating-point operations are emulated in
-  software.
+
+* Floating-point operations are emulated in software unless LLVM is told to
+  enable FPU support (either by using an `eabihf` target, specifying a
+  `target-cpu` with FPU support, or using a `target-feature` to support for a
+  specific kind of FPU)
 * Integer division is also emulated in software on some targets, depending on
-  the CPU.
-* Architectures prior to ARMv7 don't have atomic instructions.
+  the target, `target-cpu` and `target-feature`s.
+* Architectures prior to ARMv7 don't have atomic compare-and-swap instructions
```

Comment:
This bullet is incorrect in a few ways:
 - There are architectures prior to Armv7 that have atomics (like Armv6), and architectures after it that don't (like Armv6-M).
 - AArch64 has compare-and-swap instructions, but Arm does not. Arm has exclusive load+stores, which are slightly more general than CAS, and CAS can be implemented on top of them.
 - `fetch_add` is not a compare-and-swap; it unconditionally stores a new value and returns the old value. It can be implemented in terms of a compare-and-swap, but this is not how it's implemented on Arm.

Reply:
Interesting. I was going off how https://docs.rs/portable-atomic phrased things. I'll fix it!

Reply:
https://doc.rust-lang.org/stable/std/sync/atomic/index.html#portability also talks about Compare and Swap, and not Load Exclusive.

Reply:
I think I fixed it

Reply:
Ha, yeah, the wording in sync::atomic's docs is misleading. I think it probably comes from our x86-dominated history, where indeed CAS is the fundamental atomic primitive that everything sits on top of.